app/shared/inputs/FileDropZone.js

```diff
@@ -0,0 +1,147 @@
+'use client';
+
+import { useState, useRef } from 'react';
+import { CameraAltRounded } from '@mui/icons-material';
+import { useSnackbarState } from '@shared/utils/Snackbar';
+import Overline from '@shared/typography/Overline';
+
+// TODO: add more file types
+const ACCEPTED_FILE_TYPES = {
+  image: ['image/png', 'image/jpeg', 'image/gif'],
+};
+
+/**
+ * @typedef {Object} FileDropZoneProps
+ * @property {string[]} fileTypes Types of files to accept
+ * @property {(selected: File) => void} onChange Callback to send newly selected file
+ * @property {string} className css classes to add
+ */
+
+/**
+ * Input element for drag + dropping file
+ * Currently only for images
+ * @param {FileDropZoneProps} props
+ * @returns
+ */
+export default function FileDropZone({
```

Comment:
why not use a lib for that?
https://www.npmjs.com/package/react-drag-drop-files